### PR TITLE
Expose pass timings

### PR DESCRIPTION
**Pull request by sklam**
```python
llvm.set_time_passes(True)
<compile something>
report = llvm.report_and_reset_timings() # return str
llvm.set_time_passes(False)
```

### CODE REVIEW

File: docs/source/user-guide/binding/pass_timings.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+.. currentmodule:: llvmlite.binding
+
+.. function:: set_time_passes(enable)
+
+    Enable or disable pass timers.
+
+
+.. function:: report_and_reset_timings()
+
+    Returns pass timings report and reset LLVM internal timers.
```
Comment (stuartarchibald) [inline docs/source/user-guide/binding/pass_timings.rst:14]:
```suggestion
    Returns the pass timings report and resets the LLVM internal timers.
```

File: docs/source/user-guide/binding/pass_timings.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+============
+Pass Timings
+============
+
+.. currentmodule:: llvmlite.binding
+
+.. function:: set_time_passes(enable)
+
+    Enable or disable pass timers.
```
Comment (stuartarchibald) [inline docs/source/user-guide/binding/pass_timings.rst:9]:
```suggestion
    Enable or disable the pass timers.
```

File: llvmlite/binding/passmanagers.py [hunk 3]
```diff
@@ -53,6 +53,34 @@ def dump_refprune_stats(printout=False):
...
+def set_time_passes(enable):
+    """Enable or disable pass timers.
+
+    Parameters
+    ----------
+    enable : bool
+        Set to True to enable pass timers.
+        Set to False to disable pass timers.
+    """
+    ffi.lib.LLVMPY_SetTimePasses(c_int(enable))
```
Comment (stuartarchibald) [inline llvmlite/binding/passmanagers.py:65]: Can this be a `c_bool` and then the binding also use bool?

File: llvmlite/binding/passmanagers.py [hunk 4]
```diff
@@ -53,6 +53,34 @@ def dump_refprune_stats(printout=False):
...
 
 
+def set_time_passes(enable):
+    """Enable or disable pass timers.
+
+    Parameters
+    ----------
+    enable : bool
+        Set to True to enable pass timers.
+        Set to False to disable pass timers.
```
Comment (stuartarchibald) [inline llvmlite/binding/passmanagers.py:63]:
```suggestion
        Set to True to enable the pass timers.
        Set to False to disable the pass timers.
```

File: llvmlite/binding/passmanagers.py [hunk 5]
```diff
@@ -53,6 +53,34 @@ def dump_refprune_stats(printout=False):
                       stats.fanout_raise)
 
 
+def set_time_passes(enable):
+    """Enable or disable pass timers.
```
Comment (stuartarchibald) [inline llvmlite/binding/passmanagers.py:57]:
```suggestion
    """Enable or disable the pass timers.
```

File: llvmlite/binding/passmanagers.py [hunk 6]
```diff
@@ -53,6 +53,34 @@ def dump_refprune_stats(printout=False):
...
+    ----------
+    enable : bool
+        Set to True to enable pass timers.
+        Set to False to disable pass timers.
+    """
+    ffi.lib.LLVMPY_SetTimePasses(c_int(enable))
+
+
+def report_and_reset_timings():
+    """Returns pass timings report and reset LLVM internal timers.
```
Comment (stuartarchibald) [inline llvmlite/binding/passmanagers.py:69]:
```suggestion
    """Returns the pass timings report and resets the LLVM internal timers.
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, looks good, just few minor things to resolve. I've also tested this locally and can confirm it works!

File: llvmlite/binding/passmanagers.py (hunk 3 above)
Comment (sklam) [reply llvmlite/binding/passmanagers.py]: fixed in https://github.com/numba/llvmlite/pull/624/commits/bf2159ea486292245c0ca99d87a5bd70c55159f2

**Review by stuartarchibald — APPROVED** [review]
Thanks for the fixes, looks good.

**Pull request merged**